src/components/OneTimeCodeInput/index.tsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react'
+import styled from '@emotion/styled'
+import { fieldStyles, Help, LabelText } from '../Input'
+import { space } from '../../theme'
+import { useRef, useState } from 'react'
+import { VisuallyHidden } from '../VisuallyHidden'
+
+interface OneTimeCodeContainerProps {
+  length: number
+}
+
+interface OneTimeCodeInputProps {
+  length: number
+  id: string
+  hideLabel?: boolean
+  ariaLabel: string
+  label?: string
+  disabled?: boolean
+  helpText?: string
+  onChange?: (value: any) => void
+}
+
+const OneTimeCodeContainer = styled.section<OneTimeCodeContainerProps>`
+  display: grid;
+  grid-template-columns: repeat(${props => props.length}, 1fr);
```

Comment:
Thinking out loud: Maybe we should have a min length of 2 and max length of let's say, 8? because now it already looks a bit bad on mobile if you have 8, and even more will overflow the container. I think we can be opinionated about that, by saying that you can only have inputs of those lengths. Otherwise, the design looks really crappy.

Reply:
I agree, I think it's should be 6 to be honest